core/src/main/scala/org/apache/spark/deploy/worker/DriverWrapper.scala

```diff
@@ -66,4 +75,50 @@ object DriverWrapper {
         System.exit(-1)
     }
   }
+
+  // R or Python are not supported in cluster mode so just get jars and files for the driver
+  private def setupDependencies(loader: MutableURLClassLoader, userJar: String): Unit = {
+
+    var packagesExclusions = sys.props.get("spark.jars.excludes").orNull
+    var packages = sys.props.get("spark.jars.packages").orNull
+    var repositories = sys.props.get("spark.jars.repositories").orNull
+    val hadoopConf = new HadoopConfiguration()
+    val childClasspath = new ArrayBuffer[String]()
+    var jars = sys.props.get("spark.jars").orNull
```

Comment:
Here I think `jars` and `files` can only be remote resources, otherwise how can remote driver visit SparkSubmit local resources? So I think some defensive code may be necessary.